datafusion/core/tests/sql/cast.rs

```diff
@@ -0,0 +1,111 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use crate::sql::execute_to_batches;
+use arrow::record_batch::RecordBatch;
+use datafusion::assert_batches_eq;
+use datafusion::error::Result;
+use datafusion::prelude::SessionContext;
+
+async fn execute_sql(sql: &str) -> Vec<RecordBatch> {
+    let ctx = SessionContext::new();
+    execute_to_batches(&ctx, sql).await
+}
+
+#[tokio::test]
+async fn cast_tinyint() -> Result<()> {
+    let actual = execute_sql("SELECT cast(10 as tinyint)").await;
+    let expected = vec![
+        "+-------------------------+",
+        "| CAST(Int64(10) AS Int8) |",
```

Comment:
Once https://github.com/apache/arrow-datafusion/pull/3222 is merged, the results will no longer show the `CAST` expression in the field name, so it would be good to actually check the type of the column being returned here.

Here is an example from a similar test:

```
  let actual = execute_to_batches(&ctx, sql).await;
    // array decimal(10,6) +  scalar decimal(20,0) => decimal(21,6)
    assert_eq!(
        &DataType::Decimal128(27, 6),
        actual[0].schema().field(0).data_type()
    );
```

Reply:
This is much better, thanks!